src/createBlock.ts

```diff
@@ -0,0 +1,147 @@
+    // Check if to_mid, from_mid is included in above set.
+    // TODO;
+    return bcOps;
+}
+
+/**
+ * TODO.
+ */
+function validateBcOp(bcOp) {
+    return true;
+}
+
+/**
+ * Build a Blockchain Operation from a Firestore Operation. Returns undefined if the operation is invalid.
+ */
+function buildBcOpFromFsOp(
+        fsOp, index: number): Operation|void {
+    const opCode = fsOp.get('op_code');
+    const operationDataBuilder = bcOpDataBuilders[opCode];
+    if (!operationDataBuilder) {
+        return undefined;
+    }
+
+    const bcOp = {
+        sequence: index,
+        op_code: opCode,
+        creator_mid: fsOp.get('creator_mid'),
+        data: operationDataBuilder(fsOp.get('data')),
+    };
+    return validateBcOp(bcOp) ? bcOp : undefined;
+}
+
+/**
+ * Return a list of all valid unapplied operations from Firestore.
+ */
+async function getBlockOperations(): Promise<Operation[]> {
+    const unappliedFsOps = await get(
+        filters.applied(false)(
+            operationsCollection()));
+
+    let i = 0;
+    const blockchainOperations = unappliedFsOps.map(
+        (op) => buildBcOpFromFsOp(op, i++));
+    return blockchainOperations.filter(x => x) as Operation[];
+}
+
+/**
+ * Assumes the blockchain list is sorted by sequence number. Returns the last sequence number + 1.
+ */
+function getNextBlockSequenceNumber(blockchain: Block[]): number {
+    return blockchain[blockchain.length - 1].data.sequence + 1;
+}
+
+/**
+ * Assumes the blockchain list is sorted by sequence number. Returns the hash of the last block.
+ */
+function getLastBlockHash(blockchain: Block[]): string {
+    return blockchain[blockchain.length - 1].metadata.multiHash;
+}
+
+/**
+ * Create an IpfsBlock with all valid unapplied operations from Firestore.
+ */
+async function createIpfsBlock(): Promise<IpfsBlock> {
+    const blockchain = await getBlockchain();
+    const blockOperations = await getBlockOperations();
+    const validOperations = filterBcOpsAgainstBlockchain(blockchain, blockOperations);
+
+    return {
+        sequence: getNextBlockSequenceNumber(blockchain),
+        origin_created: undefined,
+        version: BLOCKCHAIN_VERSION_NO,
+        prev_version_block: undefined,
+        prev_hash: getLastBlockHash(blockchain),
+        operations: validOperations,
+    };
+}
+
+/**
+ * Compute an IPFS MultiHash over the provided data.
+ * TODO.
+ */
+function computeMultiHash(data: string) {
+    return 'multihash';
+}
+
+/**
+ * Return Block as string.
+ */
+function formatData(data) {
+    return JSON.stringify(data, null, 4);
+}
+
+/**
+ * Create a Block with all valid unapplied operations from Firestore.
+ */
+async function createBlock(): Promise<Block> {
+    const data = await createIpfsBlock();
+    const formattedData = formatData(data);
+    return {
+        metadata: {
+            multiHash: computeMultiHash(formattedData),
+            timeStr: '',
```

Comment:
Why are these empty?

Reply:
These values are only defined once a block has been recorded in the stellar blockchain. Perhaps I should let them be optional parameters.